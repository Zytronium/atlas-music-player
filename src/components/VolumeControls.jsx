import { Volume2 } from "lucide-react";

export default function VolumeControls() {
  return (
    <div className="mb-1 flex flex-row">
      <Volume2 className="size-5 fill-black" />
      <input
        type="range"
        defaultValue={75}
        className="ml-2 w-full accent-gray-500"
      />
    </div>
  );
}
