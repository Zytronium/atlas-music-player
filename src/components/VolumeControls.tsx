import { Volume2 } from "lucide-react";
import { JSX } from "react";

type Props = {
  loading?: boolean;
};

export default function VolumeControls(props: Props): JSX.Element {
  return (
    <div className="mb-1 flex flex-row">
      <Volume2
        className={`${props.loading ? "opacity-50" : ""} text-text fill-text dark:text-primary dark:fill-primary size-5`}
      />
      <input
        type="range"
        defaultValue={75}
        readOnly={props.loading}
        className={`${props.loading ? "pointer-events-none opacity-50" : ""} accent-primary ml-2 w-full`}
      />
    </div>
  );
}
