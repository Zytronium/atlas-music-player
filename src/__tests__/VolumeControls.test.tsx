import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import VolumeControls from "../components/VolumeControls";
import { afterAll, afterEach, beforeAll } from "vitest";
import { server } from "./mocks";

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("VolumeControls - default (no props)", () => {
  const { container } = render(<VolumeControls />);
  expect(container).toMatchSnapshot();
});

test("VolumeControls - loading state (controls visually disabled)", () => {
  const { container } = render(<VolumeControls loading />);
  expect(container).toMatchSnapshot();
});

test("VolumeControls - with setVolume callback provided", () => {
  const { container } = render(<VolumeControls setVolume={() => {}} />);
  expect(container).toMatchSnapshot();
});

test("VolumeControls - loading true with setVolume provided", () => {
  const { container } = render(<VolumeControls loading setVolume={() => {}} />);
  expect(container).toMatchSnapshot();
});

test("VolumeControls - explicitly passing setVolume as undefined", () => {
  const { container } = render(<VolumeControls setVolume={undefined} />);
  expect(container).toMatchSnapshot();
});
