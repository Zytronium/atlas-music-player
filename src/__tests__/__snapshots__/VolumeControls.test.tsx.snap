// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls - default (no props) 1`] = `
<div>
  <div
    class="mb-1 flex flex-row"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-volume2 lucide-volume-2  text-text fill-text dark:text-primary dark:fill-primary size-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class=" accent-primary ml-2 w-full"
      type="range"
      value="50"
    />
  </div>
</div>
`;

exports[`VolumeControls - explicitly passing setVolume as undefined 1`] = `
<div>
  <div
    class="mb-1 flex flex-row"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-volume2 lucide-volume-2  text-text fill-text dark:text-primary dark:fill-primary size-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class=" accent-primary ml-2 w-full"
      type="range"
      value="50"
    />
  </div>
</div>
`;

exports[`VolumeControls - loading state (controls visually disabled) 1`] = `
<div>
  <div
    class="mb-1 flex flex-row"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-volume2 lucide-volume-2 opacity-50 text-text fill-text dark:text-primary dark:fill-primary size-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class="pointer-events-none opacity-50 accent-primary ml-2 w-full"
      readonly=""
      type="range"
      value="50"
    />
  </div>
</div>
`;

exports[`VolumeControls - loading true with setVolume provided 1`] = `
<div>
  <div
    class="mb-1 flex flex-row"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-volume2 lucide-volume-2 opacity-50 text-text fill-text dark:text-primary dark:fill-primary size-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class="pointer-events-none opacity-50 accent-primary ml-2 w-full"
      readonly=""
      type="range"
      value="50"
    />
  </div>
</div>
`;

exports[`VolumeControls - with setVolume callback provided 1`] = `
<div>
  <div
    class="mb-1 flex flex-row"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-volume2 lucide-volume-2  text-text fill-text dark:text-primary dark:fill-primary size-5"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class=" accent-primary ml-2 w-full"
      type="range"
      value="50"
    />
  </div>
</div>
`;
